pvlib/pvsystem.py

```diff
@@ -421,6 +421,91 @@ def sapm_effective_irradiance(self, poa_direct, poa_diffuse,
             poa_direct, poa_diffuse, airmass_absolute, aoi,
             self.module_parameters, reference_irradiance=reference_irradiance)
 
+    def first_solar_spectral_correction(self, pw, airmass_absolute):
+        
+        """
+        Use the :py:func:`first_solar_spectral_correction` function to 
+        calculate the spectral loss modifier.
+
+        Parameters
+        ----------
+        pw : array-like
+            atmospheric precipitable water (cm).
+    
+        airmass_absolute : array-like
+            absolute (pressure corrected) airmass.
+
+        Returns
+        -------
+        modifier: array-like
+            spectral mismatch factor (unitless) which is can be multiplied
+            with broadband irradiance reaching a module's cells to estimate
+            effective irradiance, i.e., the irradiance that is converted to
+            electrical current.
+        """        
+
+        if self.module_parameters['fs_spectral_coefficients']:
+            coefficients = self.module_parameters['fs_spectral_coefficients']
+            module_type = None
+        else:
+            module_type = self._infer_cell_type()
```

Comment:
module vs. cell is unfortunate here. No objection to what you've done, but we might need a separate discussion to try to clean up the terminology within the library.

Reply:
Here I'm inclined to break the API and change the keyword argument from module_type to cell_type.  module_type is only used by the first_solar function.